### ISSUE TITLE
NVDA Tool - Sub grid section on a form doesn’t recognize the total number records

### ISSUE THREAD

**Issue by HeenaAttar** [nvaccess/nvda#9575]
We are using NVDA Tool for checking how accessibility works in our project and found that for subgrid section on form NVDA tool doesn't announce total number of records even though it has below structure.
-	DIV with role=grid
                o	Div with role=columnheader
                        	Div with role=gridcell
              o	Div with role=row
                        	Div with role=gridcell

Another finding was NVDA tool reads proper number of records of grid if it has same structure with table tag

Steps to reproduce:
1. Create a standalone page with ARIA table (Div elements with above hierarchy).

2. Use navigation using tab and arrow key till subgrid section

Actual behavior:
Does not announce total number of records of grid.

Expected behavior:

NVDA Tool should announce total number of records of grid 

System configuration
NVDA installed/portable/running from source:
NVDA installed as app on desktop and turned on

NVDA version:
2019.1.1

Windows version:
Windows 10

Name and version of other software in use when reproducing the issue:
Other information about your system:
Version of Chrome: 74.0.3729.131 (Official Build) (64-bit)
Version of Microsoft Edge : 44.18362.1.0

Other questions
Does the issue still occur after restarting your PC?
Yes

Have you tried any other versions of NVDA? If so, please report their behaviors.
No

**Comment by Adriani90** [comment]
@HeenaAttar could you please upload a minimal test case or a website where we can reproduce this issue? Thanks for reporting!

**Comment by HeenaAttar** [comment]
[Subgrid.zip](https://github.com/nvaccess/nvda/files/3191756/Subgrid.zip)

@Adriani90 please find standalone page which have similar structure in our application where we can reproduce this issue. On initial load or refresh on standalone page NVDA says table has 2 rows and 2 columns, where as in our application with same dom structure it does not says table with respective rows and columns. Secondly, when we navigate via tab or arrow keys in NVDA tool do not read total number of records in grid.

**Comment by Adriani90** [comment]
@HeenaAttar testing on this sample page in Crhome 74 and Edge 44 as well as Firefox 66.0.5 as follows:
1. Open NVDA
2. Open the sample page
3. In browse mode, navigate with tab and shift+tab and also with arrow keys from the first grid to the second one and then back into the first grid.
4. In focus mode, Navigate with tab and shift+tab from a grid into the other.
Actual browse mode: for me NVDA says "table with 2 rows and 2 columns" after loading the page and also when leaving the first grid and entering the second grid with arrow keys or tab. While navigating through the grid with tab, arrow keys or ctrl+alt+arrow keys, NVDA reports "row one column one", "row one column two" etc. The number of records is somehow deductable when you enter the table because NVDA says "table with two rows and two columns". This is intended behavior. The number of records should not be reported everytime when pressing tab or arrow keys in a grid because this would be to verbose.

Actual focus mode: NVDA says just "table" when entering a grid without announcing the number of rows and columns.
Expected focus mode: NVDA should report the number of rows and columns when entering a grid.

cc: @jcsteh maybe you have a suggestion regarding this. It is also reproducible in Firefox.

**Comment by Adriani90** [comment]
cc: @MarcoZehe

**Comment by MarcoZehe** [comment]
Column headers must also be contained in a row container.

**Comment by feerrenrut** [comment]
The sample starts with:
```html
<div role="grid">
			<div role="row">		
				<div tabindex="0" role="columnheader">				
					 First Name				
				</div>			
				<div tabindex="0" role="columnheader">				 
					 Last Name
				</div>
			</div> 
...
```
@HeenaAttar I think @MarcoZehe  is suggesting this is an authoring error. The provided sample is missing a `rowgroup` see https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Table_Role

**Comment by jcsteh** [comment]
No; a rowgroup is not required, though it is allowed. The [markup](https://github.com/nvaccess/nvda/issues/9575#issuecomment-617731745) does include a row; the [original issue comment](https://github.com/nvaccess/nvda/issues/9575#issue-444921394) just neglected to include the row in the described structure.

@Adriani90 is correct: the table counts are reported in browse mode, but not focus mode. I'm not entirely convinced reporting these in focus mode is useful/necessary, but I'll leave that to others. I think that's the bug that's being reported here, though it's unclear from @HeenaAttar's description because we don't have the exact keystrokes used. The tab key was mentioned, which would switch to focus mode for a grid.

**Comment by Adriani90** [comment]
I am reopening this issue, it is imo quite important to report the number of rows and columns when entering the table also in focus mode, especially in web apps where focus mode is recommended to navigate (i.e. google drive etc.). Such apps might provide very useful shortcuts which makes the usability much more efficient than in browse mode. Thus, when tabbing into a table, I would expect NVDA to report the same as in browse mode.

**Comment by feerrenrut** [comment]
Ah, sorry for the confusion there.
